.github/workflows/pytest.yml

```diff
@@ -0,0 +1,43 @@
+name: Pytest Checks
+
+on: 
+  pull_request:
+    paths:
+      - "tuning/*"
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.7
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10.12'
+
+      - name: Install dev dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements-dev.txt
+
+      - name: Set virtual env
+        run: |
+          python -m venv .venv
+          source .venv/bin/activate
+
+      - name: Install tuner dependencies
+        run: |
+          pip install -r tuning/tuner_requirements/requirements.txt
+          python -m pip install iree-compiler
```

Comment:
We could install the nightly version:
```
python -m pip install \
  --find-links https://iree.dev/pip-release-links.html \
  --upgrade \
  iree-compiler
```

This may be better if the stable is not updated very often. What do you think, @ScottTodd?

Reply:
Stable is currently updated around monthly. I'd prefer to direct downstream projects/users to use the stable releases and use that as a forcing function to get our release process running more regularly/smoothly... but most of our projects are using the nightly releases anyway. Up to you.

Reply:
will go with nightly version